vllm/model_executor/weight_utils.py

```diff
@@ -31,13 +31,18 @@
 
 
 def enable_hf_transfer():
-    """enforces activation of hf_transfer if installed."""
-    try:
-        # enable hf hub transfer if available
-        import hf_transfer  # type: ignore # noqa
-        huggingface_hub.constants.HF_HUB_ENABLE_HF_TRANSFER = True
-    except ImportError:
-        pass
+    """automatically activates hf_transfer
+    """
+    if "HF_HUB_ENABLE_HF_TRANSFER" in os.environ:
+        #  no additional logic, use default behaviour.
+        return
+    else:
+        try:
+            # enable hf hub transfer if available
+            import hf_transfer  # type: ignore # noqa
+            huggingface_hub.constants.HF_HUB_ENABLE_HF_TRANSFER = True
+        except ImportError:
+            pass
```

Comment:
Simplify? (sorry if this is overly pedantic!)
```suggestion
    if "HF_HUB_ENABLE_HF_TRANSFER" not in os.environ:
        try:
            # enable hf hub transfer if available
            import hf_transfer  # type: ignore # noqa
            huggingface_hub.constants.HF_HUB_ENABLE_HF_TRANSFER = True
        except ImportError:
            pass
```